components/core/src/Grep.cpp

```diff
@@ -567,19 +570,22 @@ std::optional<Query> Grep::process_raw_query(
         // Compute logtypes and variables for query
         auto matchability = generate_logtypes_and_vars_for_subquery(
                 archive,
-                processed_search_string,
+                search_string_for_sub_queries,
                 query_tokens,
-                query.get_ignore_case(),
+                ignore_case,
                 sub_query
         );
         switch (matchability) {
             case SubQueryMatchabilityResult::SupercedesAllSubQueries:
-                // Clear all sub-queries since they will be superceded by this sub-query
-                query.clear_sub_queries();
-
                 // Since other sub-queries will be superceded by this one, we can stop processing
                 // now
-                return query;
+                return Query{
+                        search_begin_ts,
+                        search_end_ts,
+                        ignore_case,
+                        processed_search_string,
+                        {}
+                };
```

Comment:
Is the convention here to explicitly state the type when using {} initializer for returns?